.github/workflows/core.yml

```diff
@@ -545,13 +570,73 @@ jobs:
           path: core
       - uses: ./.github/actions/environment
       - run: nix-shell --run "poetry run make -C core coverage"
-      # TODO fail if too little
       - uses: actions/upload-artifact@v3
         with:
           name: core-coverage-${{ matrix.model }}
           path: core/htmlcov
           retention-days: 7
 
+  core_ui_master:
+    name: UI diff from master
+    runs-on: ubuntu-latest
+    needs:
+      - core_click_test
+      - core_persistence_test
+      - core_device_test
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v3
+        with:
+          name: ui-records
+      - uses: ./.github/actions/environment
+      - run: "for F in screens_*.tar; do tar xvf $F; done"
+      - run: nix-shell --run "poetry run python -m tests.ui_tests.reporting master-diff TT TR"
+      - run: mv tests/ui_tests/reports/master_diff .
+      - run: |
+          chmod 600 ./ssh-id
+          nix-shell -p rsync -p openssh --run "rsync -av -e 'ssh -i ./ssh-id -o StrictHostKeyChecking=off' master_diff ui@trezorui.martinmilata.cz:ci-storage/${{ github.run_id }}"
+
+  core_ui_comment:
+    name: Post comment with UI diff URLs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+      - run: |
+          git fetch origin main
+          git diff --quiet origin/main...HEAD -- tests/ui_tests/fixtures.json || echo "FIXTURES_CHANGED=$?" >> $GITHUB_OUTPUT
+        id: check-fixtures-changed
+      - uses: ./.github/actions/ui-comment
+        if: ${{ steps.check-fixtures-changed.outputs.FIXTURES_CHANGED == '1' }}
+
+  core_ui_deploy:
```

Comment:
This doesnt work on github yet, is this something that we want to have or is it enough what you did with the ui tests in this pr?

Reply:
Is there no deploy runner? I think this is needed for master diffs to work, though as long as we keep the GitLab pipelines running we can convert it later.